File: 30-days-of-code/java/Day11.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Day11 {

    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        int arr[][] = new int[6][6];
        int hourGlassRowCol = 3;
        int hourGlassSize = arr.length - hourGlassRowCol + 1;
        int maximum = Integer.MIN_VALUE;
        for(int i = 0; i < 6; i++){
            for(int j = 0; j < 6; j++){
                arr[i][j] = in.nextInt();
            }
        }
        for (int i = 0; i < hourGlassSize; i++) {
            for(int j = 0; j < hourGlassSize; j++){
                int sum = arr[i][j] + arr[i][j + 1] + arr[i][j + 2]
                        + arr[i+ 1][j + 1]
                        + arr[i + 2][j] + arr[i+ 2][j + 1] + arr[i + 2][j + 2];
                if (sum > maximum) maximum = sum;
            }
        }
        System.out.println(maximum);
    }
}

File: 30-days-of-code/go/Day11.go
package main

import "fmt"

func make2DArray(x int, y int) [][]int {
	arr := make([][]int, y)
	for i := range arr {
		arr[i] = make([]int, x)
	}
	return arr
}

func main() {
	const (
		MAXUNIT = ^uint(0)
		MINUNIT = 0
		MAXINT  = int(MAXUNIT >> 1)
		MININT  = -MAXINT - 1
	)

	arr := make2DArray(6, 6)
	hourGlassRowCol := 3
	hourGlassSize := len(arr) - hourGlassRowCol + 1
	maximum := MININT
	for i := 0; i < 6; i++ {
		for j := 0; j < 6; j++ {
			var n int
			fmt.Scanf("%d", &n)
			arr[i][j] = n
		}
	}
	for i := 0; i < hourGlassSize; i++ {
		for j := 0; j < hourGlassSize; j++ {
			sum := arr[i][j] + arr[i][j+1] + arr[i][j+2] + arr[i+1][j+1] + arr[i+2][j] + arr[i+2][j+1] + arr[i+2][j+2]

			if sum > maximum {
				maximum = sum
			}
		}
	}
	fmt.Println(maximum)
}
